acceptance/analyzer_test.go

```diff
@@ -740,91 +727,254 @@ func testAnalyzer(t *testing.T, when spec.G, it spec.S) {
+									containerName,
+									copyDir,
+									ctrPath("/layers"),
+									analyzeImage,
+									h.WithFlags(
+										"--env", "DOCKER_CONFIG=/docker-config",
+										"--network", registryNetwork,
+										"--env", "CNB_PLATFORM_API="+apiString,
+									),
+									h.WithArgs(
+										ctrPath(analyzerPath),
+										"-cache-image",
+										authRegCacheImage,
+										"some-image",
+									),
+								)
+
+								assertLogsAndRestoresCacheMetadata(t, copyDir, output)
+							})
+						})
+					})
+
+					when("no auth registry", func() {
+						var authRegCacheImage, noAuthRegCacheImage string
+
+						it.Before(func() {
+							metadata := minifyMetadata(t, filepath.Join("testdata", "analyzer", "cache_image_metadata.json"), platform.CacheMetadata{})
 
-					assertLogsAndRestoresCacheMetadata(t, copyDir, output)
+							imageName := "some-cache-image-" + h.RandString(10)
+							authRegCacheImage, _ = buildAuthRegistryImage(
+								t,
+								imageName,
+								filepath.Join("testdata", "analyzer", "cache-image"),
+								"--build-arg", "fromImage="+containerBaseImage,
+								"--build-arg", "metadata="+metadata,
+							)
+
+							noAuthRegCacheImage = noAuthRegistry.RepoName(imageName)
+						})
+
+						it.After(func() {
+							h.DockerImageRemove(t, authRegCacheImage)
+						})
+
+						it("restores cache metadata", func() {
+							h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7")
+							output := h.DockerRunAndCopy(t,
+								containerName,
+								copyDir,
+								ctrPath("/layers"),
+								analyzeImage,
+								h.WithFlags(
+									"--network", registryNetwork,
+									"--env", "CNB_PLATFORM_API="+apiString,
+								),
+								h.WithArgs(
+									ctrPath(analyzerPath),
+									"-cache-image", noAuthRegCacheImage,
+									"some-image",
+								),
+							)
+
+							assertLogsAndRestoresCacheMetadata(t, copyDir, output)
+						})
+					})
+				})
+
+				when("cache directory case", func() {
+					it("restores cache metadata", func() {
+						h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.7")) >= 0, "Platform API >= 0.7")
+						output := h.DockerSeedRunAndCopy(t,
+							containerName,
+							cacheFixtureDir, ctrPath("/cache"),
+							copyDir, ctrPath("/layers"),
+							analyzeImage,
+							h.WithFlags(
+								"--env", "CNB_PLATFORM_API="+apiString,
+							),
+							h.WithArgs(
+								ctrPath(analyzerPath),
+								"-cache-dir", ctrPath("/cache"),
+								"some-image",
+							),
+						)
+
+						assertLogsAndRestoresCacheMetadata(t, copyDir, output)
+					})
 				})
 			})
 		})
-	})
 
-	when("Platform API < 0.5", func() {
 		when("layers path is provided", func() {
 			it("uses the group path at the working directory and writes analyzed.toml at the working directory", func() {
+				h.SkipIf(t, api.MustParse(apiString).Compare(api.MustParse("0.6")) >= 0, "Platform API < 0.5")
```

Comment:
I don't think this skip is actually necessary (at least, the tests still pass when I took it out)

Reply:
Yeah, this was a straight conversion. I didn't want it to fail and it was explicitly beneath the `0.5` `when`.